operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/cache/PassThroughResourceCache.java

```diff
@@ -0,0 +1,73 @@
+package io.javaoperatorsdk.operator.processing.cache;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.javaoperatorsdk.operator.processing.KubernetesResourceUtils;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PassThroughResourceCache<R extends CustomResource> {
+
+  private static final Logger log = LoggerFactory.getLogger(PassThroughResourceCache.class);
+
+  private ResourceCache resourceCache;
+  private MixedOperation<R, KubernetesResourceList<R>, Resource<R>> client;
+  private final ObjectMapper objectMapper;
+  private Map<String, CustomResourceIdForClient> customResourceNameId = new ConcurrentHashMap<>();
```

Comment:
Why do we need this?

Reply:
@metacosm this is WIP for now, just did a draft release so it's easyer to see chnages. There is now a more fundamnetal issue, how we propagate UUID-s from event sources. Basically you cannot read CR from the server by UUID, so will have to  make there more changes also to EventSource structure.